tasks/tasks.py

```diff
@@ -79,4 +83,113 @@ async def check_events(bot: commands.Bot) -> None:
             else:
                 logger.info(
                     f"No new upcoming events in the next 24 hours for guild {guild}."
-                )
\ No newline at end of file
+                )
+                
+@tasks.loop(minutes=5)
+async def check_concluded_proposals_task(bot: commands.Bot):
+    """
+    This function is a task that runs every 5 minutes. It checks ongoing proposals and processes them if they have ended.
+    If a proposal has ended, and meets the criteria of passing, a snapshot proposal will be created.
+
+    Parameters:
+    bot (commands.Bot): The bot instance containing the ongoing_votes attribute.
+
+    Returns:
+    None
+    """
+    logger.info("Checking to see if proposals have ended")
+    try:
+        keys_to_remove = []  # Initialize list to store keys for removal
+
+        for proposal_id, proposal_data in bot.ongoing_votes.items():
+            if time.time() < proposal_data["end_time"]:
+                continue
+
+            channel = bot.get_channel(int(proposal_data["channel_id"]))
+            if not channel:
+                logger.error(
+                    f"Unable to find the channel with id: {proposal_data['channel_id']}"
+                )
+                continue
+
+            thread = channel.get_thread(int(proposal_id))
+            if not thread:
+                logger.error(
+                    f"Unable to find the thread with id: {proposal_id} in the channel: {channel.name}"
+                )
+                continue
+
+            message = await thread.fetch_message(int(proposal_id))
+            if not message:
+                logger.error(
+                    f"Unable to find the message with id: {proposal_id} in the thread: {thread.name}"
+                )
+                continue
```

Comment:
:) perhaps consider a loop or method chaining here

Reply:
> :) perhaps consider a loop or method chaining here

Will look at this today 👍🏼

Reply:
> :) perhaps consider a loop or method chaining here

Cannot do method chaining as I am using fetch_message which is async

Reply:
Have reduced the if / elif statements by using a dict and a for loop. 

I am not so sure I have addressed your other point of using a loop, there is already a for loop in use.

Reply:
i meant for the whole if else blocks but all good